src/Symfony/Component/Templating/PhpEngine.php

```diff
@@ -330,6 +330,9 @@ public function escape($value, $context = 'html')
      */
     public function setCharset($charset)
     {
+        if ('UTF8' === $charset = strtoupper($charset)) {
```

Comment:
why do we need that? Shouldn't it be deprecated if it is for bc? Looks like a hack.